scripts/upload_test_results/upload_to_github.py

```diff
@@ -0,0 +1,175 @@
+    parser.add_argument('-t', '--title_msg', help="Title message", required = True)
+    parser.add_argument('-m', '--commit_msg', help="Commit message", required = True)
+
+    return parser.parse_args()
+
+def git_repo_update(ver_folder_name, repo_dir, upstream_repo, origin_repo, branch_git_title, commit_name, remote_name, origin_name):
+    # Create if necessary or update test_results git repo on your computer
+    if not os.path.exists(repo_dir):
+        empty_repo = Repo.init(repo_dir)
+        upstream = empty_repo.create_remote(remote_name, upstream_repo)
+
+        logging.info('Creating new repository, please wait ...')
+        assert upstream.exists()
+        assert upstream == empty_repo.remotes.upstream == empty_repo.remotes[remote_name]
+        upstream.fetch()
+
+        # Setup a local tracking branch of a remote branch
+        empty_repo.create_head('master', upstream.refs.master)  # create local branch "master" from remote "master"
+        empty_repo.heads.master.set_tracking_branch(upstream.refs.master)  # set local "master" to track remote "master
+        empty_repo.heads.master.checkout()  # checkout local "master" to working tree
+
+        empty_repo.create_remote(origin_name, origin_repo)
+        origin = Repo(os.path.join(repo_dir))
+        logging.info('Push new repo updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+    else:
+        logging.info("Repository already created, let's update master...")
+        # Use created repo to store results
+        origin = Repo(os.path.join(repo_dir))
+        upstream = Repo(os.path.join(repo_dir))
+        origin.git.checkout('master')
+        logging.info('Pull updates from the upstream ...')
+        upstream.git.pull()
+        logging.info('Push updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+
+    try:
+        # Create a new branch for daily master results
+        new_branch_name = branch_git_title + commit_name
+        origin.create_head(new_branch_name)
+        logging.info('Checkout to the new branch %s', new_branch_name)
+        origin.git.checkout(new_branch_name)
+    except Exception:
+        logging.info('Branch for the current commit already existing, skip step')
+
+    # create full git directory path where your results will be stored
+    git_dir_path = os.path.join(repo_dir, 'results', ver_folder_name)
+    logging.info('Destination directory path to store file in Git %s', git_dir_path)
+
+    if not os.path.exists(git_dir_path):
+        os.mkdir(git_dir_path)
+        logging.info('Path created %s', git_dir_path)
+    else:
+        logging.info('Path already existing')
+
+    return git_dir_path, new_branch_name, origin
+
+def add_push_to_github(git_dir_path, origin, commit_msg, title_msg, github_user, target_repo, github_token):
+    # Add your test results to the repo, commit them and then push them to create PR on Github
+    for j in glob.glob("{}/*.xml".format(file_path)):
```

Comment:
When we run sanitycheck with different targets as below:
```
./zephyr/scripts/sanitycheck --platform $target -O $target_out
```
Zephyr sanitycheck will generate 3 xml files, `sanitycheck.xml`, `sanitycheck_report.xml` and `$target.xml` in different directories. So can you set parameter `--res_dir`  like sanitycheck  [--testcase-root](https://github.com/zephyrproject-rtos/zephyr/blob/94721b316f634c3a4c24f7a839ab87363248d0d2/scripts/sanitycheck#L515), then scan `res_dir` to retrieve XML files, verify XML files (filename, file content, etc) before uploading ?

Reply:
Files should be verified by a developer manually, at least their content. 

When I did the development of that script, all my `target.xml` results were stored in one directory. I did that because our internal CI system put all target files into one directory. It explains reason that current script has the same approach idea to take all `target.xml` files from one directory.
My script was not developed to take `target.xml` file for each target platform from its `sanity-out` folder separately.

For each run on the platform(target) will be generated `sanity-out` folder. When you run sanitycheck one by one for multiple platforms you will have `sanity-out`` sanity-out.1` `sanity-out.2` `sanity-out.3` and so on.
After that, you have to take all your `target.xml` files from that folders, and put them into common folder for example `zephyr-v2.3.0-1307-gf014ba1ff1`
With that folder will work my script, it will read `target.xml` files one by one. Process of moving `target.xml` files from `sanity-out` directory to the common folder should be made on your side.

For example, our Intel Daily Sanitycheck system runs sanitycheck for each platform, and then generates path for the new commit like `master/master-4cf1832fa8/junit4TestRail/` In folder `junit4TestRail` it stores ONLY `target.xml` files. They are moving to the directory `junit4TestRail` from `sanity-out` folders using bash script.

I think necessary to discuss that point. Thank you for your input. @nashif @erwango @hakehuang @PerMac I'm looking forward your comments about that.

Reply:
I edited README file about that actions, please check it. "1. Setup a common folder for target.xml files"

Reply:
@maksimmasalski just to add some clarity on our use of `sanitycheck` script. As @wangnuannuan mentioned we run 1 instance of the `sanitycheck` script for each platform. Moreover we do it every time in a clean environment, basically building from scratch. The reason for that is we may schedule each platform to its own "executor node" and:
1. Utilize compute power of multiple servers if we talk about simulated platforms like QEMU & nSIM
2. Use special machines with attached real boards only for building and running those platforms which are attached to it
So basically we're trying to be flexible here.

Now going back to the discussion above due to explained above approach we don't have multiple `sanity-out` folders, so it's not a problem for us really. And I think real question is how to submit those separate results so that they look like from 1 snapshot. Though given we control from which commit to build (all those separate executions are building from exactly the same commit of Zephyr) it should be a straight-forward solution - to group results by the commit hash (if that's not done already - didn't check that, sorry).

Reply:
For example, we at Intel are grouping results by a main script. Command to run script which I wrote here is integrated into main script, after all results grouped into 1 folder.